sputnikdao-factory2/src/factory_manager.rs

```diff
@@ -0,0 +1,167 @@
+
+use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
+use near_sdk::json_types::Base58CryptoHash;
+use near_sdk::serde_json;
+use near_sdk::{env, sys, AccountId, Balance, CryptoHash, Gas};
+
+/// Gas spent on the call & account creation.
+const CREATE_CALL_GAS: Gas = Gas(75_000_000_000_000);
+
+/// Gas allocated on the callback.
+const ON_CREATE_CALL_GAS: Gas = Gas(10_000_000_000_000);
+
+/// Leftover gas after creating promise and calling update.
+const GAS_UPDATE_LEFTOVER: Gas = Gas(20_000_000_000_000);
+
+const NO_DEPOSIT: Balance = 0;
+
+/// Factory manager that allows to store/load contracts by hash directly in the storage.
+/// Uses directly underlying host functions to not load any of the data into WASM memory.
+#[derive(BorshSerialize, BorshDeserialize)]
+pub struct FactoryManager {}
+
+impl FactoryManager {
+    /// Store contract from input.
+    pub fn store_contract(&self) {
+        unsafe {
+            // Load input into register 0.
+            sys::input(0);
+            // Compute sha256 hash of register 0 and store in 1.
+            sys::sha256(u64::MAX as _, 0 as _, 1);
+            // Check if such blob already stored.
+            assert_eq!(
+                sys::storage_has_key(u64::MAX as _, 1 as _),
+                0,
+                "ERR_ALREADY_EXISTS"
+            );
+            // Store value of register 0 into key = register 1.
+            sys::storage_write(u64::MAX as _, 1 as _, u64::MAX as _, 0 as _, 2);
+            // Load register 1 into blob_hash.
+            let blob_hash = [0u8; 32];
+            sys::read_register(1, blob_hash.as_ptr() as _);
+            // Return from function value of register 1.
+            let blob_hash_str = serde_json::to_string(&Base58CryptoHash::from(blob_hash))
+                .unwrap()
+                .into_bytes();
+            sys::value_return(blob_hash_str.len() as _, blob_hash_str.as_ptr() as _);
+        }
+    }
+
+    /// Delete code from the contract.
+    pub fn delete_contract(&self, code_hash: Base58CryptoHash) {
+        let code_hash: CryptoHash = code_hash.into();
+        env::storage_remove(&code_hash);
+    }
+
+    /// Get code for given hash.
+    pub fn get_code(&self, code_hash: Base58CryptoHash) {
+        let code_hash: CryptoHash = code_hash.into();
+        unsafe {
+            // Check that such contract exists.
+            assert_eq!(
+                sys::storage_has_key(code_hash.len() as _, code_hash.as_ptr() as _),
+                1,
+                "Contract doesn't exist"
+            );
+            // Load the hash from storage.
+            sys::storage_read(code_hash.len() as _, code_hash.as_ptr() as _, 0);
+            // Return as value.
+            sys::value_return(u64::MAX as _, 0 as _);
+        }
+    }
+
+    /// Forces update on the given contract.
+    /// Contract must support update by factory for this via permission check.
+    pub fn update_contract(
```

Comment:
`update_existing_dao_contract`?

Reply:
nope, this is generic library - will move to near sdk later